mux/scripts/smoke-attach.py

```
@@ -21,12 +22,58 @@
 BIN = os.environ.get("CMUX_MUX_BIN", "target/debug/cmux-mux")
 SESSION = f"smoke-attach-{os.getpid()}"
-SOCK = os.path.join(
-    os.environ.get("TMPDIR", "/tmp"), f"cmux-mux-{os.getuid()}", f"{SESSION}.sock"
-)
+SOCK = None
+CONTROL_SOCKET_RE = re.compile(r"control socket at (.+)$")
 MARKER = f"reattach-marker-{os.getpid()}"
 
 
+def fallback_socket_path():
+    base = os.environ.get("XDG_RUNTIME_DIR") or os.environ.get("TMPDIR") or "/tmp"
```

Ruff (0.15.20)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)